### PR TITLE
New unit tests, NPC ships/fleets, say() debugging, misc bugfixes

### PR DESCRIPTION
**Pull request by caelaislinn**
See commit log. 
:cl: CaelAislinn
wip: Endgame UNSC fleet should be fairly stronger now. 
bugfix: potential bug fix for chat bug
bugfix: fix faction lock on ONI base
/:cl:

### CODE REVIEW

**Comment by caelaislinn** [comment]
#2523

File: code/modules/halo/misc/bumpstairs_oni.dm [hunk 1]
```diff
@@ -4,7 +4,9 @@
 /obj/structure/bumpstairs/road/gem_oni
 	id_self = "gem_oni"
 	id_target = "oni_gem"
+	faction_restrict = "UNSC"
```
Comment (X0-11) [inline code/modules/halo/misc/bumpstairs_oni.dm:7]: wait this was the issue?

File: code/modules/halo/overmap/combat_npc_ship.dm [hunk 2]
```diff
@@ -207,17 +208,19 @@
...
 	radio_channel = RADIO_COV
 	ship_datums = list(/datum/npc_ship/cov_patrol)
 	available_ship_requests = newlist(/datum/npc_ship_request/halt/cov,/datum/npc_ship_request/fire_on_target/cov,/datum/npc_ship_request/control_fleet/cov,/datum/npc_ship_request/add_to_fleet/cov,/datum/npc_ship_request/give_control/cov)
 
 /obj/effect/overmap/ship/npc_ship/combat/covenant/medium_armed
+	icons_pickfrom_list = list('code/modules/halo/icons/overmap/kig_missionary.dmi')
 	projectiles_to_fire = list(/obj/item/projectile/overmap/pulse_laser = 0.2 SECONDS,/obj/item/projectile/overmap/plas_torp = 0.5 SECONDS)
 
 /obj/effect/overmap/ship/npc_ship/combat/covenant/heavily_armed
+	icons_pickfrom_list = list('code/modules/halo/icons/overmap/SDV.dmi')
```
Comment (X0-11) [inline code/modules/halo/overmap/combat_npc_ship.dm:223]: i think the sdv is too big to be used as an npc ship sprite

File: code/modules/halo/misc/bumpstairs_oni.dm (hunk 1 above)
Comment (caelaislinn) [reply code/modules/halo/misc/bumpstairs_oni.dm:7]: Yep. I was initially planning to have it be auto copied from the creator `bumpstairs/road` but hardcoding them like this is actually easier

File: code/modules/halo/overmap/combat_npc_ship.dm (hunk 2 above)
Comment (caelaislinn) [reply code/modules/halo/overmap/combat_npc_ship.dm:223]: This is the heavy NPC ship which has the MAC/energy projector, so it should look dangerous

Comment (X0-11) [reply code/modules/halo/overmap/combat_npc_ship.dm:223]: i often feel like the bigger the shipsprite the easier it is to kill

**Review by BDpuffy420 — APPROVED** [review]

**Comment by caelaislinn** [comment]
Here @X0-11 if the silence bug happens you set `var/debug_say = 1` on your test mob and make sure you have hear debug log enabled, then try talking.

**Pull request merged**